packages/astro/src/core/base-pipeline.ts

```diff
@@ -45,7 +45,7 @@ export abstract class Pipeline {
 		/**
 		 * Used for `Astro.site`.
 		 */
-		readonly site = manifest.site
+		readonly site = manifest.site ? new URL(manifest.site) : undefined,
```

Comment:
You are right, it will throw. However, I don't think it needs extra code to handle the case. If it fails at runtime, it would also necessarily fail in dev and in build.

Reply:
Isn't site already validated at https://github.com/withastro/astro/blob/5fe98599a3fc6f912995d94ce783c32c06c0b44f/packages/astro/src/core/config/schema.ts#L95

Reply:
Oh right, it is validated!